webui/react/src/shared/utils/data.ts

```diff
@@ -9,7 +9,8 @@ export const isBoolean = (data: unknown): data is boolean => typeof data === 'bo
 export const isDate = (data: unknown): data is Date => data instanceof Date;
 export const isMap = (data: unknown): boolean => data instanceof Map;
 export const isNullOrUndefined = (data: unknown): data is null | undefined => data == null;
-export const isNumber = (data: unknown): data is number => typeof data === 'number';
+export const isNumber = (data: unknown): data is number =>
+  typeof data === 'number' && !Number.isNaN(data);
```

Comment:
checked in at #saas-web-sharing about this. but apparently `isNumber(Number.NaN) === true` unless you do this. "Nan" means "not a number"